### ISSUE TITLE
Error: Cannot read property 'ctx' of undefined

### ISSUE THREAD

**Issue by cyr-x** [ngxs-labs/immer-adapter#164]
If you use the `ImmutableContext ` decorator  together with an `Action` decorator that is subscribing to an Action that is also subscribed to from another State you get the following error:
```
ERROR TypeError: Cannot read property 'ctx' of undefined
 at push.../../node_modules/@ngxs-labs/immer-adapter/fesm5/ngxs-labs-immer-adapter.js.ImmutableStateContext.getState
// ...
```

An example to reproduce this:

```TS

@State<{}>({
  name: 'state1',
  defaults: {}
})
export class State1
{
  @Action(Action1)
  @ImmutableContext()
  public onMyAction1({ getState }: StateContext<{}>)
  {
    getState();
  }
}

@State<{}>({
  name: 'state2',
  defaults: {}
})
export class State2
{
  @Action(Action1)
  @ImmutableContext()
  public onMyAction1({ getState }: StateContext<{}>)
  {
    getState();
  }
}
```
[**STACKBLITZ**](https://stackblitz.com/edit/angular-innjv1)

**Comment by splincode** [comment]
This is bug, alternative use

```ts
public onMyAction1(ctx: StateContext<{}>) {
  ctx.getState()
}
```

**Comment by cyr-x** [comment]
Can confirm that using `ctx.getState()` is working as expected. Will use that for now thx.

**Comment by splincode** [comment]
we need more tests for everything, you can send pull request for help me

**Comment by splincode** [comment]
fixed v3.0.2
